### ISSUE TITLE
Typing Problem

### ISSUE THREAD

**Issue by Uter1007** [apollostack/angular2-apollo#237]
Hi,

I've setuped the angularclass/ng2 stack with apollo but I get the following typing errors -> could someone provide me more infos: 

```
[at-loader] Checking finished with 7 errors         
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:75                            
    Property destructuring pattern expected.        
                                                    
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:85                            
    ',' expected.                                   
                                                    
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:86                            
    Parameter declaration expected.                 
                                                    
[at-loader] node_modules\angular2-apollo\build\src\A
ngular2Apollo.d.ts:17:77                            
    Generic type 'ApolloQueryResult' requires 1 type
 argument(s).                                       
                                                    
[at-loader] node_modules\angular2-apollo\build\src\A
ngular2Apollo.d.ts:18:61                            
    Generic type 'ApolloQueryResult' requires 1 type
 argument(s).                                       
                                                    
[at-loader] node_modules\angular2-apollo\build\src\A
ngular2Apollo.d.ts:19:50                            
    Generic type 'ApolloQueryResult' requires 1 type
 argument(s).                                       
```

**Comment by Uter1007** [comment]
just saw the following link: https://github.com/apollostack/apollo-client/issues/1164

**Comment by kamilkisiela** [comment]
Yes, it's a known issue. We made it by mistake, sorry for that :(

**Comment by kamilkisiela** [comment]
About 

```
[at-loader] Checking finished with 7 errors         
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:75                            
    Property destructuring pattern expected.        
                                                    
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:85                            
    ',' expected.                                   
                                                    
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:86                            
    Parameter declaration expected.
```

It's a thing because `apollo-client` requires TypeScript 2.1. So to fix that issue you can update TypeScript in your project. And yes, Angular doesn't support it yet but I've checked and all should be okay. We still waiting for angular/angular#13294 to be released

**Comment by kamilkisiela** [comment]
Also we're trying to make it compatible with TS 2.0 which is a good news!

**Comment by kamilkisiela** [comment]
@Uter1007 
```
[at-loader] node_modules\angular2-apollo\build\src\A
ngular2Apollo.d.ts:17:77                            
    Generic type 'ApolloQueryResult' requires 1 type
 argument(s).                                       
                                                    
[at-loader] node_modules\angular2-apollo\build\src\A
ngular2Apollo.d.ts:18:61                            
    Generic type 'ApolloQueryResult' requires 1 type
 argument(s).                                       
                                                    
[at-loader] node_modules\angular2-apollo\build\src\A
ngular2Apollo.d.ts:19:50                            
    Generic type 'ApolloQueryResult' requires 1 type
 argument(s).   
```

Has been fixed with `0.9.0-rc.5`

```
[at-loader] Checking finished with 7 errors         
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:75                            
    Property destructuring pattern expected.        
                                                    
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:85                            
    ',' expected.                                   
                                                    
[at-loader] node_modules\apollo-client\lib\src\data\
readFromStore.d.ts:26:86                            
    Parameter declaration expected.
```

Those will be fixed by apollostack/apollo-client#1171 soon.

So there'd no reason not to use TS 2.0.

**Comment by kamilkisiela** [comment]
Fixed in #243
